### PR TITLE
#2 done

### PR DESCRIPTION
**Pull request by kowaliks**
#2 done

### CODE REVIEW

File: travelrec/similarity.py [hunk 1]
```diff
@@ -33,17 +33,30 @@ def construct_filters(self, words):
...
-                if similarity is not None and similarity > 0.8:
-                    top_words.append(feature)
+                for a in wn.synsets(word):
+                    for b in wn.synsets(feature):
+                        try:
+                            similarity = a.path_similarity(b)
+                            if similarity is not None and similarity > best_similarity:
+                                best_synonym = feature
+                                best_similarity = similarity
+                        except:
```
Comment (rzepinskip) [inline travelrec/similarity.py]: Specify error type for except block

File: travelrec/similarity.py [hunk 2]
```diff
@@ -33,17 +33,30 @@ def construct_filters(self, words):
...
+                            print(f"can't comapre {a} and {b}")
+
+            if best_similarity > 0.8:
+                top_words.append(best_synonym)
+            else:
+                top_words.append(None)
```
Comment (rzepinskip) [inline travelrec/similarity.py]: I always prefer returning empty list instead of list of None

File: tests/test_similarity.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+from travelrec.similarity import ActvitySimilarity, ClimateSimilarity, GeoFeaturesSimilarity
+
+import unittest
```
Comment (rzepinskip) [inline tests/test_similarity.py]: I prefer `pytest` due to `pytest.mark.parametrize` and `pytest.fixture` decorators. You can use them to provide list of examples to test and have each example as separate test. I am not sure whether there is a equivalent feature in `unittest`.

File: tests/test_similarity.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    def test_climate_similarity(self):
+        climateSim = ClimateSimilarity()
+        words = ['mild', 'warm']#, 'medium', 'hot']
+        correct_synonym = ['mild', 'warm']#, 'mild', 'warm']
+
+        filters = climateSim.construct_filters(words)
+
+        self.assertEqual(len(words), len(filters), 'number of filters is wrong')
+
+        for i in range(0, len(words)):
```
Comment (rzepinskip) [inline tests/test_similarity.py]: Default lower bound of range is 0, so you can use `for i in range(len(words)):`. Even more pythonic version would be `for synonym, filter, word in zip(correct_synonym, filters, words):`

File: tests/test_similarity.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        results = [[], ["T.MT", "T.MTS"], ["H.LK", "H.LKS"], ["T.MT", "T.MTS", "H.LK", "H.LKS"]]
+
+        for i in range(0, len(examples)):
+            example = examples[i]
+            nlp_extended = NlpExtender(example)
+            nouns = nlp_extended.nouns()
+
+            similarities = geoFeaturesSim.construct_filters(nouns)
+
+            self.assertEqual(len(similarities), len(results[i]))
```
Comment (rzepinskip) [inline tests/test_similarity.py]: To compare lists use `assertListEqual` - https://stackoverflow.com/a/31832447
In pytest there is simple overload for `assert` - https://stackoverflow.com/a/46933938

File: travelrec/similarity.py [hunk 6]
```diff
@@ -33,17 +33,30 @@ def construct_filters(self, words):
...
-                    top_words.append(feature)
+                for a in wn.synsets(word):
+                    for b in wn.synsets(feature):
+                        try:
+                            similarity = a.path_similarity(b)
+                            if similarity is not None and similarity > best_similarity:
+                                best_synonym = feature
+                                best_similarity = similarity
+                        except:
+                            print(f"can't comapre {a} and {b}")
```
Comment (rzepinskip) [inline travelrec/similarity.py]: Misspelling

File: travelrec/nlp_extender.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import en_core_web_sm
+
+class NlpExtender():
```
Comment (rzepinskip) [inline travelrec/nlp_extender.py]: Rename to `ProcessedStatement`?

**Review by rzepinskip — CHANGES_REQUESTED** [review]

File: travelrec/similarity.py (hunk 2 above)
Comment (kowaliks) [reply travelrec/similarity.py]: Maybe I did not understand what you intended to be the output of this method. Isn't it a list of closest synonyms of given words in same order? In pipeline.py line 43 you use only first element of returned array.
What is your idea about that?

File: tests/test_similarity.py (hunk 3 above)
Comment (kowaliks) [reply tests/test_similarity.py]: Ok, it sounds great, but I had problems with imports from travelrec if tests are in separate directory

> ModuleNotFoundError: No module named 'travelrec'

I didn't find any easy solution, howewer I'm new to python. If you have expirience with that, could you please configure one simple test using imports from travelrec?

File: travelrec/similarity.py (hunk 2 above)
Comment (rzepinskip) [reply travelrec/similarity.py]: I thought of it as keywords detected for specified input words. I used only first one for demonstration purposes, we should use all of them.

File: tests/test_similarity.py (hunk 3 above)
Comment (rzepinskip) [reply tests/test_similarity.py]: You have to install our package in dev mode `pip install -e .`

**Review by rzepinskip — APPROVED** [review]

**Pull request merged**